### PR TITLE
feat: add gasprice to tx object namespace

### PR DESCRIPTION
**Pull request by skellet0r**
### What I did

Add `gasprice` to `tx` namespace

### How to verify it

Check test

### Description for the changelog

feat: add gasprice to tx object namespace

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: docs/constants-and-vars.rst [hunk 1]
```diff
@@ -25,6 +25,7 @@ Name                 Type             Value
 ``msg.sender``       ``address``      Sender of the message (current call)
 ``msg.value``        ``uint256``      Number of wei sent with the message
 ``tx.origin``        ``address``      Sender of the transaction (full call chain)
+``tx.gasprice``      ``uint256``      Gas price of current transaction in wei
```
Comment (fubuloubu) [inline docs/constants-and-vars.rst:28]: What is this set to post-EIP1559?

**Review by fubuloubu — APPROVED** [review]

File: docs/constants-and-vars.rst (hunk 1 above)
Comment (skellet0r) [reply docs/constants-and-vars.rst:28]: For 1559 (type 2) txs, it'll just match `BASEFEE`, but for legacy txs it'll be whatever gasprice was set when the tx was built.

https://eips.ethereum.org/EIPS/eip-1559#gasprice

Comment (skellet0r) [reply docs/constants-and-vars.rst:28]: Mainly useful for EVM chains that don't support LONDON yet

Comment (fubuloubu) [reply docs/constants-and-vars.rst:28]: perfect

**Pull request merged**